### DISCUSSION TITLE
attach-policy returning Policy not found error

### DISCUSSION THREAD

**Discussion by jimenaCabrejas** [aws/aws-cli#7005] [Q&A] [answered]
Hello,

I'm trying to attach a policy to a cognito Identity Id using the command aws iot attach-policy. I have verified that the policy I'm trying to attach is correctly created, I can see this:
```
$aws iam get-policy --policy-arn arn:aws:iam::xxxxx:policy/IoTFullPolicy
{
    "Policy": {
        "PolicyName": "IoTFullPolicy",
        "Arn": "arn:aws:iam::xxxxx:policy/IoTFullPolicy",
        "Path": "/",
        "DefaultVersionId": "v1",
        "AttachmentCount": 5,
        "PermissionsBoundaryUsageCount": 0,
        "IsAttachable": true,
        "CreateDate": "2022-06-02T11:54:42Z",
        "UpdateDate": "2022-06-02T11:54:42Z"
    }
}
```
But when I try to attach it I get the following error:
```
$aws iot attach-policy --policy-name "IoTFullPolicy" --target "eu-west-1:xxxxxxxxxx"

An error occurred (ResourceNotFoundException) when calling the AttachPolicy operation: Policy not found
```

Any idea why this is happening?

**Comment by tim-finnigan** [comment, marked as answer]
Hi @jimenaCabrejas thanks for reaching out. Here is the `aws iot attach-policy` documentation for reference: https://docs.aws.amazon.com/cli/latest/reference/iot/attach-policy.html

When using the `--target` parameter it looks like you need the full ARN. For example:
`--target "arn:aws:iot:us-west-2:123456789012:thinggroup/LightBulbs`

**Reply by jimenaCabrejas** [reply]
Hi! Thank you for your answer. I was following this Amplify tutorial https://docs.amplify.aws/lib/pubsub/getting-started/q/platform/js/#step-1-create-iam-policies-for-aws-iot
Following this I only know my Cognito Identity ID. Where can I see the full ARN of my Cognito Identity ID?

Edit: I just found it! Thank you

**Reply by eddiemuhoro** [reply]
@jimenaCabrejas  Where did you find the full ARN then?

**Comment by KamleshAtTray** [comment]
@jimenaCabrejas What full ARN for the Cognito Idenitity Id did you use ? I am facing the same issue of Policy Not Found using the code. But when I use the CLI command the policy gets attached successfully the Cognito Identity ID ( without full ARN ).
